pyCGM_Single/tests/test_pycgmIO.py

```diff
@@ -0,0 +1,870 @@
+          {'A': [2, 3, 4]}]
+        )
+    ])            
+    def test_createMotionDataDict_accuracy(self, labels, data, expected_result):
+        """
+        This function tests pycgmIO.createMotionDataDict(labels,data),
+        where labels is a list of label names and data is a 2d list 
+        or numpy array of coordinates corresponding to labels.
+
+        We test cases where the values in data are lists and numpy arrays.
+        We test cases where there are more labels than data coordinates.
+        We test cases where there are more data coordinates than labels.
+        """
+        result = pycgmIO.createMotionDataDict(labels, data)
+        np.testing.assert_equal(result, expected_result)
+
+    @pytest.mark.parametrize("motiondata, expected_labels, expected_data", [
+        (
+         [{'A': [1, 2, 3], 'B': [4, 5, 6]},
+          {'A': [2, 3, 4], 'B': [5, 6, 7]}],
+         ['A', 'B'],
+         np.array([[[1, 2, 3],[4, 5, 6]],[[2, 3, 4],[5, 6, 7]]])
+        ),
+        (
+         [{'A': np.array([1, 2, 3]), 'B': np.array([4, 5, 6])},
+          {'A': np.array([2, 3, 4]), 'B': np.array([5, 6, 7])}],
+         ['A', 'B'],
+         np.array([[[1, 2, 3],[4, 5, 6]],[[2, 3, 4],[5, 6, 7]]])
+        ),
+        (
+         [{'A': np.array([1, 2, 3]), 'B': np.array([4, 5, 6])},
+          {'A': np.array([2, 3, 4])}],
+         ['A', 'B'],
+         np.array([[[1, 2, 3],[4, 5, 6]],[[2, 3, 4],[2, 3, 4]]])
+        ),
+        (
+         [{'B': np.array([4, 5, 6])},
+          {'A': np.array([2, 3, 4])}],
+         ['B'],
+         np.array([[[4, 5, 6]],[[2, 3, 4]]])
+        )
+    ])
+    def test_splitMotionDataDict_accuracy(self, motiondata, expected_labels, expected_data):
+        """
+        This function tests pycgmIO.splitMotionDataDict(motiondata),
+        where motiondata is a list of dictionaries of motion capture data.
+        This function splits the motiondata into a tuple of labels, data.
+
+        We tests cases where values are lists or numpy arrays.
+        We demonstrate unexpected behavior that the function produces when
+        keys are not present in every dictionary of motiondata.
+        """
+        result_labels, result_data = pycgmIO.splitMotionDataDict(motiondata)
+        assert result_labels == expected_labels
+        np.testing.assert_equal(result_data, expected_data)
+    
+    @pytest.mark.parametrize("motiondata", [
+        ([{'A': [1, 2]}]),
+        ([{'A': []}]),
+        ([{'A': [1, 2, 3, 4, 5]}]),
+        ([{'A': [[1, 2, 3],
+               [4, 5, 6]]}])
+    ])
+    def test_splitMotionDataDict_exceptions(self, motiondata):
+        """
+        We test that if the dictionary values in motiondata are not
+        1d arrays of 3 elements, an exception is raised.
+        """
+        with pytest.raises(Exception):
+            pycgmIO.splitMotionDataDict(motiondata)
+
+    @pytest.mark.parametrize("labels, data, expected_result", [
+        (
```

Comment:
Can you put these parenthesis on the same lines as the functions, this is a bit annoying to read.